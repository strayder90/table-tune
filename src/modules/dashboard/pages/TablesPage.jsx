import React from 'react';

const TablesPage = () => {

    return (
        <>
            <h1>Welcome to the Home Page!</h1>
        </>
    );
};

export default TablesPage;
