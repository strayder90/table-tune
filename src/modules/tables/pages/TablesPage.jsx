import React from 'react';

import Floor from '@modules/tables/components/Floor.jsx';

const TablesPage = () => {
    return (
        <div className='--page-wrapper'>
            <Floor />
        </div>
    );
};

export default TablesPage;
