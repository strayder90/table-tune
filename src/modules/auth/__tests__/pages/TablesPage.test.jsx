import React from 'react';
import '@testing-library/jest-dom';
import {render, screen} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import {userEvent} from '@testing-library/user-event';

import TablesPage from '@modules/dashboard/pages/TablesPage.jsx';

const mockNavigate = vi.fn();
vi.doMock('react-router-dom', () => ({
    ...vi.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('TablesPage test suite', () => {
    test('renders without crashing', () => {
        const {container} = render(
            <MemoryRouter>
                <TablesPage/>
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();

        const header = screen.getByRole('heading', {name: /Welcome to the Home Page!/i});
        expect(header).toBeInTheDocument();

        const logoutButton = screen.getByRole('button', {name: /Logout/i});
        expect(logoutButton).toBeInTheDocument();
    });

    test('clicking logout button clears localStorage and redirects to login', async () => {
        localStorage.setItem('isAuthenticated', 'true');

        render(
            <MemoryRouter>
                <TablesPage/>
            </MemoryRouter>
        );

        expect(localStorage.getItem('isAuthenticated')).toBe('true');

        const logoutButton = screen.getByRole('button', {name: /Logout/i});

        await userEvent.click(logoutButton);

        expect(localStorage.getItem('isAuthenticated')).toBeNull();
    });
});
