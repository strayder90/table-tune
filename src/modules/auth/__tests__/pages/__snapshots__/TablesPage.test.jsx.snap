// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TablesPage test suite > renders without crashing 1`] = `
<div>
  <h1>
    Welcome to the Home Page!
  </h1>
</div>
`;
